### PR TITLE
Some cleanup of mount create API test

### PR DESCRIPTION
**Pull request by dnephin**
While debugging a test failure I made some changes to this test to make it easier to debug.

The API test no longer uses the CLI.

### CODE REVIEW

**Comment by vieux** [comment]
LGTM

**Review by boaz0 — APPROVED** [review]
LGTM

**Comment by andrewhsu** [comment]
I've [seen failure on DockerSuite.TearDownTest and subsequent panic](https://jenkins.dockerproject.org/job/Docker-PRs-WoW-RS1/16839/console):
```
----------------------------------------------------------------------
FAIL: check_test.go:99: DockerSuite.TearDownTest

check_test.go:100:
    testEnv.Clean(c)
c:/gopath/src/github.com/docker/docker/vendor/github.com/stretchr/testify/assert/assertions.go:226:
    t.Errorf("\r" + getWhitespaceString() + labeledOutput(content...))
... Error: 
                          
	Error Trace:	clean.go:84

	            	
			clean.go:35

	            	
			check_test.go:100

	            	
			asm_amd64.s:514

	            	
			value.go:434

	            	
			value.go:302

	            	
			check.go:764

	            	
			check.go:672

	            	
			asm_amd64.s:2197

	Error:      	Received unexpected error:

	            	Error response from daemon: No such container: afa6ae5bc8596cc6885f503ad28315b118eb8bf1fb4eec841f57793eb5ad2b44

	Messages:   	failed to remove afa6ae5bc8596cc6885f503ad28315b118eb8bf1fb4eec841f57793eb5ad2b44



----------------------------------------------------------------------
PANIC: docker_cli_restart_test.go:296: DockerSuite.TestRestartAutoRemoveContainer

... Panic: Fixture has panicked (see related PANIC)

----------------------------------------------------------------------
```

**Comment by andrewhsu** [comment]
Kicking windowsRS1 build again: https://jenkins.dockerproject.org/job/Docker-PRs-WoW-RS1/16843/

**Comment by dnephin** [comment]
That clean error is weird, the code should ignore "not found" errors. I'll have to debug that, but I don't think it's related to this. It's related to #34656

File: integration-cli/docker_api_containers_test.go [hunk 1]
```diff
@@ -1921,7 +1924,23 @@ func (s *DockerSuite) TestContainersAPICreateMountsCreate(c *check.C) {
 		{mounttypes.Mount{Type: "volume", Target: destPath + slash}, types.MountPoint{Driver: volume.DefaultDriverName, Type: "volume", RW: true, Destination: destPath}},
 		{mounttypes.Mount{Type: "volume", Target: destPath, Source: "test1"}, types.MountPoint{Type: "volume", Name: "test1", RW: true, Destination: destPath}},
 		{mounttypes.Mount{Type: "volume", Target: destPath, ReadOnly: true, Source: "test2"}, types.MountPoint{Type: "volume", Name: "test2", RW: false, Destination: destPath}},
-		{mounttypes.Mount{Type: "volume", Target: destPath, Source: "test3", VolumeOptions: &mounttypes.VolumeOptions{DriverConfig: &mounttypes.Driver{Name: volume.DefaultDriverName}}}, types.MountPoint{Driver: volume.DefaultDriverName, Type: "volume", Name: "test3", RW: true, Destination: destPath}},
+		{
+			mounttypes.Mount{
```
Comment (cpuguy83) [inline integration-cli/docker_api_containers_test.go]: Trying to spot the diff here.

Comment (dnephin) [reply integration-cli/docker_api_containers_test.go]: I didn't change any of the values, just fixed the formatting so that it's readable now

Comment (cpuguy83) [reply integration-cli/docker_api_containers_test.go]: Why just this one?

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**